File: src/MixNetFx/Facesso.EntityModel/FacessoEntitiesPartial.vb
Imports System
Imports System.Data.Objects
Imports System.Data.Objects.DataClasses
Imports System.Data.EntityClient
Imports System.ComponentModel
Imports System.Xml.Serialization
Imports System.Runtime.Serialization

Imports Facesso

Partial Public Class FacessoEntities

    Public Shared Function GetFacessoEntityString() As String
        Dim sqlString = FacessoGeneric.SQLConnectionString
        Dim providerString = "System.Data.SqlClient"

        If sqlString.IndexOf("MultipleActiveResultSets") < 0 Then
            sqlString &= ";MultipleActiveResultSets=True"
        End If

        Dim eb As New EntityConnectionStringBuilder(sqlString)
        eb.Provider = providerString
        eb.Metadata = "metadata=res://*/FacessoModel.csdl|res://*/FacessoModel.ssdl|res://*/FacessoModel.msl"
        Return eb.ToString

    End Function

End Class

File: src/NetFxCSharp/Facesso.EntityModel/FacessoEntitiesPartial.cs
using Facesso;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.EntityClient;
using System.Data.Objects;
using System.Data.Objects.DataClasses;
using System.Diagnostics;
using System.Linq;
using System.Runtime.Serialization;
using System.Threading.Tasks;
using System.Xml.Linq;
using System.Xml.Serialization;

namespace Facesso.EntityModel
{
    public partial class FacessoEntities
    {
        public static string GetFacessoEntityString()
        {
            var sqlString = FacessoGeneric.SQLConnectionString;
            var providerString = "System.Data.SqlClient";
            if (sqlString.IndexOf("MultipleActiveResultSets") < 0)
            {
                sqlString += ";MultipleActiveResultSets=True";
            }

            EntityConnectionStringBuilder eb = new EntityConnectionStringBuilder(sqlString);
            eb.Provider = providerString;
            eb.Metadata = "metadata=res://*/FacessoModel.csdl|res://*/FacessoModel.ssdl|res://*/FacessoModel.msl";
            return eb.ToString();
        }
    }
}